test/unit/config/index.js

```diff
@@ -0,0 +1,291 @@
+                    config: {
+                        username: 'config_username',
+                        apiToken: 'default_token',
+                        url: 'default_url'
+                    }
+                });
+
+                const config = new Config('./docpub.config');
+
+                expect(config.username).to.be.equal('argv_username');
+            });
+
+            it('should override option in config with ENV variable prefixed with `docpub_`', () => {
+                process.env['docpub_username'] = 'env_username';
+                stubConfig_({
+                    path: './docpub.config',
+                    config: {
+                        username: 'config_username',
+                        apiToken: 'default_token',
+                        url: 'default_url'
+                    }
+                });
+
+                const config = new Config('./docpub.config');
+
+                expect(config.username).to.be.equal('env_username');
+            });
+
+            it('should use argv option, if both cli and argv options available', () => {
+                process.env['docpub_username'] = 'env_username';
+                process.argv = process.argv.concat('--username', 'argv_username');
+                stubConfig_({
+                    path: './docpub.config',
+                    config: {
+                        username: 'config_username',
+                        apiToken: 'default_token',
+                        url: 'default_url'
+                    }
+                });
+
+                const config = new Config('./docpub.config');
+
+                expect(config.username).to.be.equal('argv_username');
+            });
+
+            it('should allow to not declare options in json config if argv option available', () => {
+                process.argv = process.argv.concat('--username', 'argv_username');
+                stubConfig_({
+                    path: './docpub.config',
+                    config: {
+                        apiToken: 'default_token',
+                        url: 'default_url'
+                    }
+                });
+
+                const config = new Config('./docpub.config');
+
+                expect(config.username).to.be.equal('argv_username');
+            });
+
+            it('should allow to not declare options in json config if env option available', () => {
+                process.env['docpub_username'] = 'env_username';
+                stubConfig_({
+                    path: './docpub.config',
+                    config: {
+                        apiToken: 'default_token',
+                        url: 'default_url'
+                    }
+                });
+
+                const config = new Config('./docpub.config');
+
+                expect(config.username).to.be.equal('env_username');
+            });
+        });
+    });
+});
+
+function stubConfig_(opts) {
+    opts = _.defaults(opts || {}, {
```

Comment:
Just tried - tests started to have some strange declarations like:

```javascript
stubConfig_({
    path: './docpub.config',
    config: {
        username: undefined, <--- this one
        token: 'default_token',
        url: 'default_url'
    }
});
```
Do you think it's better than declaring required config fields all the time?